# Simpson's Paradox
Use `admission_data.csv` for this exercise.

In [3]:
import pandas as pd
dt =pd.read_csv('admission_data.csv')
dt.head()

student_id  gender      major  admitted
0       35377  female  Chemistry     False
1       56105    male    Physics      True
2       31441  female  Chemistry     False
3       51765    male    Physics      True
4       53714  female    Physics      True

### Proportion and admission rate for each gender

In [28]:
# Proportion of students that are female
female_count=dt[dt['gender'] == "female"]
total_students = len(dt)
proportion_female = len(female_count) / total_students
proportion_female

0.514

In [29]:
# Proportion of students that are male
male_count=dt[dt['gender'] == "male"]
proportion_male = len(male_count) / total_students
proportion_male

0.486

In [34]:
# Admission rate for females
female_admitted_count = female_count[female_count['admitted'] == True]
admission_rate_female = len(female_admitted_count) / len(female_count)
admission_rate_female

0.28793774319066145

In [35]:
# Admission rate for males
male_admitted_count = male_count[male_count['admitted'] == True]
admission_rate_male = len(male_admitted_count) / len(male_count)
admission_rate_male

0.48559670781893005

### Proportion and admission rate for physics majors of each gender

In [37]:
# What proportion of females students are majoring in physics?
female_physics_count=female_count[female_count['major'] == "Physics"]
proportion_female_physics=len(female_physics_count)/len(female_count)
proportion_female_physics

0.12062256809338522

In [38]:
# What proportion of male students are majoring in physics?
male_physics_count=male_count[male_count['major'] == "Physics"]
proportion_male_physics=len(male_physics_count)/len(male_count)
proportion_male_physics

0.9259259259259259

In [41]:
# Admission rate for female physics majors
female_physics_admitted_count=female_physics_count[female_physics_count['admitted'] == True]
admission_rate_female_physics=len(female_physics_admitted_count)/len(female_physics_count)
admission_rate_female_physics

0.7419354838709677

In [42]:
# Admission rate for male physics majors
male_physics_admitted_count=male_physics_count[male_physics_count['admitted'] == True]
admission_rate_male_physics=len(male_physics_admitted_count)/len(male_physics_count)
admission_rate_male_physics

0.5155555555555555

### Proportion and admission rate for chemistry majors of each gender

In [46]:
# What proportion of female students are majoring in chemistry?
female_Chemistry_count=female_count[female_count['major'] == "Chemistry"]
proportion_female_Chemistry=len(female_Chemistry_count)/len(female_count)
proportion_female_Chemistry

0.8793774319066148

In [47]:
# What proportion of male students are majoring in chemistry?
male_Chemistry_count=male_count[male_count['major'] == "Chemistry"]
proportion_male_Chemistry=len(male_Chemistry_count)/len(male_count)
proportion_male_Chemistry

0.07407407407407407

In [48]:
# Admission rate for female chemistry majors
female_Chemistry_admitted_count=female_Chemistry_count[female_Chemistry_count['admitted'] == True]
admission_rate_female_Chemistry=len(female_Chemistry_admitted_count)/len(female_Chemistry_count)
admission_rate_female_Chemistry

0.22566371681415928

In [49]:
# Admission rate for male chemistry majors
male_Chemistry_admitted_count=male_Chemistry_count[male_Chemistry_count['admitted'] == True]
admission_rate_male_Chemistry=len(male_Chemistry_admitted_count)/len(male_Chemistry_count)
admission_rate_male_Chemistry

0.1111111111111111

### Admission rate for each major

In [52]:
# Admission rate for physics majors
physics_data = dt[dt['major'] == "Physics"]
physics_admitted_count=physics_data[physics_data['admitted'] == True]
admission_rate_physics =len(physics_admitted_count)/len(physics_data)
admission_rate_physics

0.54296875

In [53]:
# Admission rate for chemistry majors
Chemistry_data = dt[dt['major'] == "Chemistry"]
Chemistry_admitted_count=Chemistry_data[Chemistry_data['admitted'] == True]
admission_rate_Chemistry =len(Chemistry_admitted_count)/len(Chemistry_data)
admission_rate_Chemistry

0.21721311475409835

### What are your conclusions?

- Proportion of students that are female is 51.4%
- Proportion of students that are male is 48.6%
- Admission rate for females is 28.7938%
- Admission rate for males is 48.5597%
- Males appears to be favored in theadmissions process
- Proportion of females with physics majors is 12.1% and with chemistry majors is 87.9%
- Proportion of males with physics majors is 92.6% and with chemistry majors is 7.4%
- Admission rate for female physics majors is 74.2% and for female chemistry majors is 22.6%
- Admission rate for male physics majors is 51.6% and for male chemistry majors is 11.1%
- Of the students applying as all majors, Females appears to be favored in the admissionsprocess
- Males tends to have more physics majors than chemistry majors?
- Females tends to have more chemistry majors than physics majors
- Chemistry major has a lower admission rate